scripts/build_dependencies/build_step_axios.js

```diff
@@ -0,0 +1,106 @@
+/**
+ * Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0.
+ */
+const fs = require("fs");
+const crypto = require('crypto');
+const utils = require('./build_utils');
+
+module.exports = {
+
+    axios: null,
+    clean_up_axios: false,
+    axios_version: "0.24.0",
```

Comment:
I am not a fan of this hard-coded version... 
- We cannot tell if any CVE or something happens to that package at that version unless manually checking the code, which could be a concern
- If the lib was existed, and it's not this version, we will not know. 

But, I cannot think of a better way😞. 
Does any other nodejs lib using the similar approach to fetch dependency dynamically? How do they deal with the version?

Reply:
I'm not sure, I would need to look around and see.

We could put it in our dev dependencies and then pull it out of the `package.json` that way. Then it would get automatically code scanned by Dependabot.

Though the dependencies here are completely isolated from the rest of the node modules by being in the `node_modules` folder of the `scripts/build_dependencies` folder. So even if there was a vulnerability, it would only be for building the source and not the output or anything a customer/consumer of the CRT would see.
Not saying it is something we shouldn't fix, but the security impact is minimal because it just gets downloaded and used only for a build.

Reply:
I am not totally against it, but if there is a better way, we should take it instead.

Reply:
I did some looking, and it looks like the general consensus is "don't do it" based on other Javascript packages and such. I got around this by putting it in the `devDependencies` and then loading the `package.json` and getting the version number from there. Now we have the advantage of getting Dependabot code scanning, can easily change the version in a single place, and it is more in line with what I have seen in other Javascript packages/modules.